cmds/record.c

```diff
@@ -263,8 +263,10 @@ static void setup_child_environ(struct opts *opts, int argc, char *argv[])
 			setenv("UFTRACE_NEST_LIBCALL", "1", 1);
 	}
 
-	if (strcmp(opts->dirname, UFTRACE_DIR_NAME))
-		setenv("UFTRACE_DIR", opts->dirname, 1);
+	char *path = realpath(opts->dirname, NULL);
+	if (mode == METHOD_ATTACH && path == NULL)
```

Comment:
I don't like declaration in the middle of code.  Also what's the `METHOD_ATTACH`?  You need to free the path as well.